Python 3.13 alpha, pathlib windows equals pathlib posix

Currently this test fails:

```
self = <tests.test_pathlib.TestComparisons testMethod=test_pure_equal>

    def test_pure_equal(self):
        """Test equivalence."""

        p1 = pathlib.PureWindowsPath('wcmatch')
        p2 = pathlib.PurePosixPath('wcmatch')

        p3 = pypathlib.PureWindowsPath('wcmatch')
        p4 = pypathlib.PurePosixPath('wcmatch')

>       self.assertTrue(p1 != p2)
E       AssertionError: False is not true

tests/test_pathlib.py:313: AssertionError
```

It should be noted that this works with the default Pathlib, so class inheritance has changed.

```py
>>> import pathlib
>>> p1 = pathlib.PureWindowsPath('path')
>>> p2 = pathlib.PurePosixPath('path')
>>> p1 == p2
False
```

We will likely have to restructure, once again, how we derive our classes from pathlib.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,7 +17,7 @@ jobs:
       max-parallel: 4
       matrix:
         platform: [ubuntu-latest, windows-latest]
-        python-version: [3.8, 3.9, '3.10', '3.11', '3.12', '3.13-dev']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12', '3.13']
         include:
           - python-version: 3.8
             tox-env: py38
@@ -33,7 +33,7 @@ jobs:
             tox-env: py313
         exclude:
           - platform: windows-latest
-            python-version: '3.13-dev'
+            python-version: '3.13'
 
     env:
       TOXENV: ${{ matrix.tox-env }}
@@ -43,15 +43,10 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - name: Set up Python ${{ matrix.python-version }}
-      if: "!endsWith(matrix.python-version, '-dev')"
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Set up development Python ${{ matrix.python-version }}
-      if: endsWith(matrix.python-version, '-dev')
-      uses: deadsnakes/action@v2.1.1
-      with:
-        python-version: ${{ matrix.python-version }}
+        allow-prereleases: true
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip setuptools tox coverage

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,7 +17,7 @@ jobs:
       max-parallel: 4
       matrix:
         platform: [ubuntu-latest, windows-latest]
-        python-version: [3.8, 3.9, '3.10', '3.11', '3.12-dev']
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12', '3.13-dev']
         include:
           - python-version: 3.8
             tox-env: py38
@@ -27,11 +27,13 @@ jobs:
             tox-env: py310
           - python-version: '3.11'
             tox-env: py310
-          - python-version: '3.12-dev'
+          - python-version: '3.12'
             tox-env: py312
+          - python-version: '3.13'
+            tox-env: py313
         exclude:
           - platform: windows-latest
-            python-version: '3.12-dev'
+            python-version: '3.13-dev'
 
     env:
       TOXENV: ${{ matrix.tox-env }}
@@ -39,7 +41,7 @@ jobs:
     runs-on: ${{ matrix.platform }}
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: Set up Python ${{ matrix.python-version }}
       if: "!endsWith(matrix.python-version, '-dev')"
       uses: actions/setup-python@v4
@@ -58,11 +60,12 @@ jobs:
         python -m tox
     - name: Upload Results
       if: success()
-      uses: codecov/codecov-action@v3
+      uses: codecov/codecov-action@v4
       with:
         file: ./coverage.xml
         flags: unittests
         name: ${{ matrix.platform }}-${{ matrix.tox-env }}
+        token: ${{ secrets.CODECOV_TOKEN }} # required
         fail_ci_if_error: false
 
   lint:
@@ -77,7 +80,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
       with:
@@ -101,7 +104,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
       with:

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -16,7 +16,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
       with:
         fetch-depth: 0
     - name: Set up Python ${{ matrix.python-version }}
@@ -40,7 +40,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-python@v4
         with:
           python-version: 3.11

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -71,7 +71,7 @@ show_error_codes = true
 [tool.ruff]
 line-length = 120
 
-select = [
+lint.select = [
     "A",    # flake8-builtins
     "B",    # flake8-bugbear
     "D",    # pydocstyle
@@ -87,7 +87,7 @@ select = [
     "PERF"  # Perflint
 ]
 
-ignore = [
+lint.ignore = [
     "E741",
     "D202",
     "D401",

--- a/wcmatch/pathlib.py
+++ b/wcmatch/pathlib.py
@@ -109,7 +109,7 @@ class PurePath(pathlib.PurePath):
         sep = ''
         name = str(self)
         if isinstance(self, Path) and name and self.is_dir():
-            sep = self._flavour.sep
+            sep = self.parser.sep if util.PY313 else self._flavour.sep
 
         return name + sep
 
@@ -227,18 +227,28 @@ class Path(pathlib.Path):
         yield from self.glob(patterns, flags=flags | _EXTMATCHBASE, limit=limit, exclude=exclude)
 
 
-class PurePosixPath(PurePath):
-    """Pure Posix path."""
+if util.PY313:
+    class PurePosixPath(pathlib.PurePosixPath, PurePath):
+        """Pure Posix path."""
 
-    _flavour = pathlib._posix_flavour if not util.PY312 else posixpath  # type: ignore[attr-defined]
-    __slots__ = ()
+        __slots__ = ()
 
+    class PureWindowsPath(pathlib.PureWindowsPath, PurePath):
+        """Pure Windows path."""
 
-class PureWindowsPath(PurePath):
-    """Pure Windows path."""
+        __slots__ = ()
+else:
+    class PurePosixPath(PurePath):  # type: ignore[no-redef]
+        """Pure Posix path."""
 
-    _flavour = pathlib._windows_flavour if not util.PY312 else ntpath  # type: ignore[attr-defined]
-    __slots__ = ()
+        _flavour = pathlib._posix_flavour if not util.PY312 else posixpath  # type: ignore[attr-defined]
+        __slots__ = ()
+
+    class PureWindowsPath(PurePath):  # type: ignore[no-redef]
+        """Pure Windows path."""
+
+        _flavour = pathlib._windows_flavour if not util.PY312 else ntpath  # type: ignore[attr-defined]
+        __slots__ = ()
 
 
 class PosixPath(Path, PurePosixPath):

--- a/wcmatch/util.py
+++ b/wcmatch/util.py
@@ -11,6 +11,7 @@ from typing import Any, Callable, AnyStr, Match, Pattern, cast
 
 PY310 = (3, 10) <= sys.version_info
 PY312 = (3, 12) <= sys.version_info
+PY313 = (3, 13) <= sys.version_info
 
 UNICODE = 0
 BYTES = 1

